Sources/VideoOptionsControlBar.swift

```diff
@@ -0,0 +1,147 @@
+    
+    @objc var repeatMode: VLCRepeatMode {
+        set {
+            rptMode = newValue
+            switch newValue {
+            case .repeatCurrentItem:
+                repeatButton.setImage(UIImage(named: "repeatOne-new"), for: .normal)
+                break
+            case .repeatAllItems:
+                repeatButton.setImage(UIImage(named: "repeat-new"), for: .normal)
+                break
+            default: // no repeat
+                repeatButton.setImage(UIImage(named: "no-repeat-new"), for: .normal)
+                break
+            }
+        }
+        
+        get {
+            return rptMode
+        }
+    }
+    
+    @objc var toggleFullScreenButton: UIButton = {
+        var toggle = UIButton(type: .system)
+        toggle.setImage(UIImage(named: "fullscreenIcon-new"), for: .normal)
+        toggle.tintColor = .white
+        //TODO: add accessability options for fullScreenButton
+        return toggle
+    }()
+    
+    @objc var selectSubtitleButton: UIButton = {
+        var subbutton = UIButton(type: .system)
+        subbutton.setImage(UIImage(named: "subtitleIcon-new"), for: .normal)
+        subbutton.tintColor = .white
+        // TODO: add accessability options for selectingSubtitleButton
+        return subbutton
+    }()
+    
+    @objc var repeatButton: UIButton = {
+        var rptButton = UIButton(type: .system)
+        rptButton.setImage(UIImage(named: "no-repeat-new"), for: .normal)
+        rptButton.tintColor = .white
+        // TODO: add accessability options for repeatButton
+        return rptButton
+    }()
+    
+    @objc var interfaceLockButton: UIButton = {
+        var interfaceLockButton = UIButton(type: .system)
+        interfaceLockButton.setImage(UIImage(named: "lock-new"), for: .normal)
+        interfaceLockButton.tintColor = .white
+        // TODO: add accessability options for orientationLockButton
+        return interfaceLockButton
+    }()
+    
+    @objc var moreOptionsButton: UIButton = {
+        var moreOptionsButton = UIButton(type: .system)
+        moreOptionsButton.setImage(UIImage(named: "moreWhite-new"), for: .normal)
+        moreOptionsButton.tintColor = .white
+        // TODO: add accessability options for moreOptionsButton
+        return moreOptionsButton
+    }()
+    
+    // MARK: Class Initializers
+    required init(coder aDecoder: NSCoder) {
+        super.init(coder: aDecoder)
+    }
+    
+    private func setupButtonTargets() {
+        toggleFullScreenButton.addTarget(self, action: #selector(toggleFullscreen), for: .touchUpInside)
+        interfaceLockButton.addTarget(self, action: #selector(toggleInterfaceLock), for: .touchUpInside)
+        repeatButton.addTarget(self, action: #selector(toggleRepeat), for: .touchUpInside)
+        selectSubtitleButton.addTarget(self, action: #selector(selectSubtitle), for: .touchUpInside)
+        moreOptionsButton.addTarget(self, action: #selector(selectMoreOptions), for: .touchUpInside)
+    }
```

Comment:
So I did do that initially, however, for some reason there was a bug where the actions were not being added to the buttons because `self` had not been created yet. There was some weird behavior happening that @Mikanbu and I couldn't understand until I did this.

Reply:
probably because they're not lazily instantiated